test/QueryPlannerTest.cpp

```diff
@@ -2130,16 +2155,7 @@ TEST(QueryPlanner, SpatialJoinLegacyPredicateSupport) {
                      PayloadAllVariables{}, S2, scan("?x", "<p>", "?y"),
                      scan("?a", "<p>", "?b")));
 
-  // For nearest neighbors, the special predicate is removed and throws an
-  // explanatory error
-  AD_EXPECT_THROW_WITH_MESSAGE(h::expect("SELECT ?y ?b WHERE {"
-                                         "?a <p> ?b ."
-                                         "?x <a> ?y ."
-                                         "?y <nearest-neighbors:1> ?b ."
-                                         "}",
-                                         ::testing::_),
-                               ::testing::ContainsRegex("Please use SERVICE"));
-
```

Comment:
I explicitly have a matcher that also checks the added warnings, you can use it here (it is also used with other tests, so you can find usage examples nearby).